kedro/io/data_catalog.py

```diff
@@ -615,15 +616,55 @@ def add_transformer(
                 )
             self._transformers[data_set_name].append(transformer)
 
-    def list(self) -> List[str]:
-        """List of ``DataSet`` names registered in the catalog.
+    def list(
+        self, regex_search: Optional[str] = None, get_objects: bool = False
```

Comment:
Do you think `regex_filter` is a more appropriate arg name? (you're the regex expert here, so up to you!)

maybe `regex_pattern`?

Reply:
I think it's fine, pattern doesn't necessarily say what you're going to do with it.